genetic_algorithms/problems/traveling_salesman_problem/move_generator.py

```diff
@@ -0,0 +1,28 @@
+import random
+from itertools import combinations
+
+
+from genetic_algorithms.models.move_generator import MoveGenerator
+from genetic_algorithms.problems.traveling_salesman_problem.state import TravelingSalesmanState
+from genetic_algorithms.problems.traveling_salesman_problem.moves import SwapEdges
+from genetic_algorithms.problems.traveling_salesman_problem.models.edge import Edge
+from typing import Generator
+
+
+class TravelingSalesmanMoveGenerator(MoveGenerator):
+    def __init__(self, depot_idx: int):
+        self._depot_idx = depot_idx
+
+    def _is_depot_start(self, a: Edge):
+        return a.start != self._depot_idx
+
+    def _is_depot_end(self, b: Edge):
+        return b.end != self._depot_idx
+
+    def random_move(self, state: TravelingSalesmanState) -> SwapEdges:
```

Comment:
To jest do poprawy na pewno, ale nie wymyśliłam, jak sensownie to ogarnąć.

Reply:
A co dokładnie Ci się nie podoba? 

Według mnie jest całkiem nieźle:
każdy algorytm dostaję coś za pomocą czego on może generować ruchy na kilka różnych sposobów (wygenerować losowy ruch z tego stanu albo wygenerować wszystkie możliwe ruchy z tego stanu) i potem sam decyduje o tym jakie ruchy on chce używać.

Reply:
Potrzeba przerobić na `Generator`.